mm2src/coins/lightning/ln_p2p.rs

```diff
@@ -0,0 +1,205 @@
+
+const TRY_RECONNECTING_TO_NODE_INTERVAL: f64 = 60.;
+const BROADCAST_NODE_ANNOUNCEMENT_INTERVAL: u64 = 600;
+
+type NetworkGossip = NetGraphMsgHandler<Arc<NetworkGraph>, Arc<dyn Access + Send + Sync>, Arc<LogState>>;
+
+pub type PeerManager =
+    SimpleArcPeerManager<SocketDescriptor, ChainMonitor, Platform, Platform, dyn Access + Send + Sync, LogState>;
+
+#[derive(Display)]
+pub enum ConnectToNodeRes {
+    #[display(fmt = "Already connected to node: {}@{}", _0, _1)]
+    AlreadyConnected(String, String),
+    #[display(fmt = "Connected successfully to node : {}@{}", _0, _1)]
+    ConnectedSuccessfully(String, String),
+}
+
+pub async fn connect_to_node(
+    pubkey: PublicKey,
+    node_addr: SocketAddr,
+    peer_manager: Arc<PeerManager>,
+) -> ConnectToNodeResult<ConnectToNodeRes> {
+    if peer_manager.get_peer_node_ids().contains(&pubkey) {
+        return Ok(ConnectToNodeRes::AlreadyConnected(
+            pubkey.to_string(),
+            node_addr.to_string(),
+        ));
+    }
+
+    match lightning_net_tokio::connect_outbound(Arc::clone(&peer_manager), pubkey, node_addr).await {
+        Some(connection_closed_future) => {
+            let mut connection_closed_future = Box::pin(connection_closed_future);
+            loop {
+                // Make sure the connection is still established.
+                match futures::poll!(&mut connection_closed_future) {
+                    std::task::Poll::Ready(_) => {
+                        return MmError::err(ConnectToNodeError::ConnectionError(format!(
+                            "Node {} disconnected before finishing the handshake",
+                            pubkey
+                        )));
+                    },
+                    std::task::Poll::Pending => {},
+                }
+
+                match peer_manager.get_peer_node_ids().contains(&pubkey) {
+                    true => break,
+                    // Wait for the handshake to complete if false.
+                    false => Timer::sleep_ms(10).await,
+                }
+            }
+        },
+        None => {
+            return MmError::err(ConnectToNodeError::ConnectionError(format!(
+                "Failed to connect to node: {}",
+                pubkey
+            )))
+        },
+    }
+
+    Ok(ConnectToNodeRes::ConnectedSuccessfully(
+        pubkey.to_string(),
+        node_addr.to_string(),
+    ))
+}
+
+pub async fn connect_to_nodes_loop(open_channels_nodes: NodesAddressesMapShared, peer_manager: Arc<PeerManager>) {
+    loop {
+        let open_channels_nodes = open_channels_nodes.lock().clone();
+        for (pubkey, node_addr) in open_channels_nodes {
+            let peer_manager = peer_manager.clone();
+            match connect_to_node(pubkey, node_addr, peer_manager.clone()).await {
+                Ok(res) => {
+                    if let ConnectToNodeRes::ConnectedSuccessfully(_, _) = res {
+                        log::info!("{}", res.to_string());
+                    }
```

Comment:
Is this log really required, it doesn't give much information about the context

Reply:
`#[derive(Display)]` is implemented for the result, so the log gives a good message that we connected to a lightning node successfully, this is important when restarting our node or when a connection drops with the other node for some time.
https://github.com/KomodoPlatform/atomicDEX-API/blob/6b2e98fda5775cd857e69a2438f80d6bc0a552f7/mm2src/coins/lightning/ln_p2p.rs#L29-L30